### PR TITLE
PG changes release version 3.x

### PR DESCRIPTION
**Pull request by vivekshankar1**
### :nut_and_bolt: Description: What code changed, and why?

This is to upgrade the postgresql version to 13 and addition of migration command.  
This PR contains all the changes that were earlier merged into this base branch.
Also fixed notifications service failure to send HTTPS requests
This also contains all the changes required to integration pipelines to solve issues occurring due to semver changes and pg changes

### :chains: Related Resources

https://chefio.atlassian.net/browse/SHIELD-22?atlOrigin=eyJpIjoiMGY1ODg2OTQ2Y2IyNDBiNzljOTc3NWM0YzgwMTdkMzciLCJwIjoiaiJ9

### :+1: Definition of Done

Pipelines should be green. 
Upgrading to this new version should have postgresql13 installed. 
Following commands should be available and working without error:
- `chef-automate post-major-upgrade migrate --data=PG`
- `chef-automate post-major-upgrade clear-data --data=PG`
- All services should be up and running
- If migrating second time it should allow with yes or no.
- If data is not migrated it should not let further upgrade
- Use all the flags for example: skip-storage-check, --force

### :athletic_shoe: How to Build and Test the Change

- Result/build.json should have milestone manifest 20220329091442
- `hab studio enter`
- `start_all_services`
- Add some dummy data
- Upgrade automate to 3.0.6 by running `chef-automate upgrade run --major`
- `rebuild components/automate-postgresql`
- `rebuild components/pg-sidecar-service`
- `rebuild components/automate-cli`
- `rebuild components/notifications-service`
- Run the migrate command `chef-automate post-major-upgrade migrate --data=PG`
- Run the clear data command `chef-automate post-major-upgrade clear-data --data=PG`

### :white_check_mark: Checklist

**All PRs** must tick these:

- [x] I have read the [CONTRIBUTING document](https://github.com/chef/automate/blob/master/CONTRIBUTING.md).
- [x] All commits signed-off for the [Developer Certification of Origin](https://github.com/chef/chef/blob/master/CONTRIBUTING.md#developer-certification-of-origin-dco).

With occasional exceptions, all PRs **from Progress employees** must tick these:

- [x] Is the code clear? *(complicated code or lots of comments--subdivide and use well-named methods, meaningful variable names, etc.)*
- [x] Consistency checked? *(user notifications, user prompts, visual patterns, code patterns, variable names)*
- [x] Repeated code blocks eliminated? *(adapt and reuse existing components, blocks, functions, etc.)*
- [x] Spelling, grammar, typos checked? *(at a minimum use `make spell` in any component directory)*
- [x] Code well-formatted? *(indents, line breaks, etc. improve rather than hinder readability)*

All PRs **from Progress employees** should tick these if appropriate:

- [ ] Tests added/updated? (all new code needs new tests)
- [x] Docs added/updated? (all customer-facing changes)

*Please add a note next to any checkbox above if you are NOT ticking it.*

### CODE REVIEW

File: components/automate-deployment/testdata/old_manifests/20220329091442.json [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+      "chef/notifications-service/1.0.0/20220315175533",
+      "chef/pg-sidecar-service/0.0.1/20220310121646",
+      "chef/sample-data-service/0.1.0/20220310121513",
+      "chef/secrets-service/1.0.0/20220310121517",
+      "chef/session-service/0.1.0/20220310121519",
+      "chef/teams-service/0.1.0/20220310121519",
+      "chef/user-settings-service/0.1.0/20220316142345",
+      "core/rsync/3.2.3/20211017043532"
+    ]
+  }
```
Comment (shaik80) [inline components/automate-deployment/testdata/old_manifests/20220329091442.json:74]: add new line

File: .expeditor/verify.pipeline.yml [hunk 2]
```diff
@@ -257,18 +257,18 @@ steps:
...
-      - cd components/automate-cli
-      - make cross
-    timeout_in_minutes: 10
-    retry:
-      automatic:
-        limit: 1
-    expeditor:
-      executor:
-        docker:
+  # - label: "[cross-compile] automate-cli"
```
Comment (vivek-yadav) [inline .expeditor/verify.pipeline.yml:260]: we can drop off the commented code

File: integration/tests/v2_to_v3_upgrade.sh [hunk 3]
```diff
@@ -0,0 +1,54 @@
+#!/bin/bash
+
+# this test script:
+# 1. deploys an older version of Automate and upgrades it to v2 using the beta CLI,
```
Comment (Dmaddu) [inline integration/tests/v2_to_v3_upgrade.sh:4]: please modify the comments in line with the test. seems these are not modified after copying

File: .expeditor/verify.pipeline.yml (hunk 2 above)
Comment (vivekshankar1) [reply .expeditor/verify.pipeline.yml]: done

File: components/automate-deployment/testdata/old_manifests/20220329091442.json (hunk 1 above)
Comment (vivekshankar1) [reply components/automate-deployment/testdata/old_manifests/20220329091442.json]: done

File: components/pg-sidecar-service/pkg/pgw/client.go [hunk 4]
```diff
@@ -594,7 +594,6 @@ func (client *Client) DeploySqitch(db, dir, user string) error {
 	sqitchArgs := command.Args(
 		"--quiet",
 		"--top-dir", dir,
-		"--engine", "pg",
```
Comment (vivek-yadav) [inline components/pg-sidecar-service/pkg/pgw/client.go:597]: Why is this removed ?

File: terraform/test-environments/modules/chef_automate_install/templates/install_chef_automate_cli.sh.tpl [hunk 5]
```diff
@@ -184,10 +184,33 @@ else
...
+    y" | chef-automate upgrade run --major --airgap-bundle /tmp/automate.aib
+
+                # NOTE: This is a hack
+                # The hack above was no longer good enough because we have a thing that needs
+                # to be updated that isn't a service
+                sleep 45
+
+                #shellcheck disable=SC2154
+                wait_for_upgrade
+                chef-automate post-major-upgrade migrate --data=PG -y
```
Comment (Dmaddu) [inline terraform/test-environments/modules/chef_automate_install/templates/install_chef_automate_cli.sh.tpl:203]: This works for the current upgrade, how about future upgrades? Does this command gracefully allow the process to continue? If this command fails, the further code won't execute

File: components/pg-sidecar-service/pkg/pgw/client.go (hunk 4 above)
Comment (vivekshankar1) [reply components/pg-sidecar-service/pkg/pgw/client.go:597]: The new version of sqitch auto-detects the db type. Engine option is not supported anymore and it gives error now

File: integration/tests/manual_upgrade.sh [hunk 6]
```diff
@@ -79,10 +80,15 @@ do_upgrade() {
...
     # Uncomment once the --version flag is on dev
-    # chef-automate upgrade run --version "$release"
-    chef-automate dev grpcurl deployment-service -- \
-        chef.automate.domain.deployment.Deployment.Upgrade -d "{\"version\": \"$release\"}"
+    echo "y
+y
+y
+y
+y" | chef-automate upgrade run --major --version "$release"
+    # chef-automate dev grpcurl deployment-service -- \
```
Comment (vivek-yadav) [inline integration/tests/manual_upgrade.sh:88]: remove commented code

File: integration/tests/upgrade.sh [hunk 7]
```diff
@@ -6,16 +6,3 @@ test_upgrades=true
 test_upgrade_inspec_profiles=(a2-deploy-integration)
 # The inspec tests don't pass if the diagnostics are run
 test_skip_diagnostics=true
-
-run_upgrade() {
```
Comment (vivek-yadav) [inline integration/tests/upgrade.sh:10]: why is this not needed now?

File: lib/platform/sys/linux.go [hunk 8]
```diff
@@ -3,13 +3,13 @@
 package sys
 
 import (
-	"bytes"
+	bytes "bytes"
+	"github.com/chef/automate/lib/proc"
```
Comment (vivek-yadav) [inline lib/platform/sys/linux.go:7]: why is automate dependency added ?

File: scripts/pgdata.sh [hunk 9]
```diff
@@ -0,0 +1,50 @@
+#!/hab/pkgs/core/bash/5.1/20210826055113/bin/bash
```
Comment (vivek-yadav) [inline scripts/pgdata.sh:1]: is this linking correct, this version might not be installed in new machines later, or even now.

File: integration/tests/iam_v1_to_v2_force_upgrade_diagnostics.sh [hunk 10]
```diff
@@ -43,3 +43,11 @@ do_deploy() {
         --skip-preflight \
         --debug
 }
+
+do_prepare_upgrade() {
+  # use latest current here
+  # shellcheck disable=SC2154
+  download_manifest_version "current" "20220329091442" "$test_manifest_dir/20220329091442.json"
```
Comment (Dmaddu) [inline integration/tests/iam_v1_to_v2_force_upgrade_diagnostics.sh:50]: As this code is same in some other files, can we write a common function and call inside do_prepare_upgrade function

File: scripts/pgdata.sh [hunk 11]
```diff
@@ -0,0 +1,50 @@
...
+
+#shellcheck disable=SC1091
+source /hab/svc/automate-postgresql/config/functions.sh
+
+mkdir -p /hab/svc/automate-postgresql/config/conf.d
+mkdir -p /hab/svc/automate-postgresql/var/pg_stat_tmp
+
+# Auto-detect pg data in the root of svc_data_path, where older versions of this plan had it
+if [[ -f "/hab/svc/automate-postgresql/data/PG_VERSION" ]]; then
+  echo "PGDATA detected in the root of the data path ( /hab/svc/automate-postgresql/data ), relocating it to /hab/svc/automate-postgresql/data/pgdata"
```
Comment (vivek-yadav) [inline scripts/pgdata.sh:29]: is this if section needed, as in new installations it will always be automate-postgresql/data/pgdata

File: scripts/pgdata13.sh [hunk 12]
```diff
@@ -0,0 +1,50 @@
+#!/hab/pkgs/core/bash/5.1/20210826055113/bin/bash
```
Comment (vivek-yadav) [inline scripts/pgdata13.sh:1]: this too, please confirm if this is pinning is needed?

File: scripts/pgdata13.sh [hunk 13]
```diff
@@ -0,0 +1,50 @@
...
+
+#shellcheck disable=SC1091
+source /hab/svc/automate-postgresql/config/functions.sh
+
+mkdir -p /hab/svc/automate-postgresql/config/conf.d
+mkdir -p /hab/svc/automate-postgresql/var/pg_stat_tmp
+
+# Auto-detect pg data in the root of svc_data_path, where older versions of this plan had it
+if [[ -f "/hab/svc/automate-postgresql/data/PG_VERSION" ]]; then
+  echo "PGDATA detected in the root of the data path ( /hab/svc/automate-postgresql/data ), relocating it to /hab/svc/automate-postgresql/data/pgdata13"
```
Comment (vivek-yadav) [inline scripts/pgdata13.sh:29]: why is `IF` section needed?

File: integration/tests/v2_to_v3_upgrade.sh (hunk 3 above)
Comment (vivekshankar1) [reply integration/tests/v2_to_v3_upgrade.sh]: done

File: integration/tests/v2_to_v3_upgrade.sh [hunk 14]
```diff
@@ -0,0 +1,54 @@
...
+        --hartifacts "$test_hartifacts_path" \
+        --override-origin "$HAB_ORIGIN" \
+        --manifest-dir "$test_manifest_path" \
+        --admin-password chefautomate \
+        --accept-terms-and-mlsa \
+        --skip-preflight \
+        --debug
+}
+
+do_upgrade() {
```
[the remark this answers is not in the document]
Comment (vivekshankar1) [reply integration/tests/v2_to_v3_upgrade.sh]: removed it

File: integration/tests/manual_upgrade.sh (hunk 6 above)
Comment (vivekshankar1) [reply integration/tests/manual_upgrade.sh]: done

File: integration/tests/iam_v1_to_v2_force_upgrade_diagnostics.sh (hunk 10 above)
Comment (vivekshankar1) [reply integration/tests/iam_v1_to_v2_force_upgrade_diagnostics.sh]: done

File: scripts/pgdata13.sh (hunk 12 above)
Comment (vivekshankar1) [reply scripts/pgdata13.sh]: changed it to /bin/bash

File: scripts/pgdata.sh (hunk 11 above)
Comment (vivekshankar1) [reply scripts/pgdata.sh]: removed it

File: scripts/pgdata13.sh (hunk 13 above)
Comment (vivekshankar1) [reply scripts/pgdata13.sh]: removed it

File: scripts/pgdata.sh (hunk 9 above)
Comment (vivekshankar1) [reply scripts/pgdata.sh]: changed to /bin/bash

File: integration/tests/upgrade.sh (hunk 7 above)
Comment (vivekshankar1) [reply integration/tests/upgrade.sh:10]: since this function is there in deployment.sh

**Review by vivek-yadav — APPROVED** [review]

**Review by shivangi-singh1 — APPROVED** [review]

**Review by shaik80 — APPROVED** [review]
